internal/shellgen/tmpl/glibc-patch.bash

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -euo pipefail
+
+declare -r pkg   # package that we're patching
+declare -r glibc # new glibc that we're patching in
+declare -r out   # nix output path that will contain the patched package
+
+# Paths to this script's dependencies set by nix.
+declare -r coreutils file findutils patchelf ripgrep
+
+# Explicitly declare the specific commands that this script depends on.
+hash -p "$coreutils/bin/cp" cp
+hash -p "$coreutils/bin/chmod" chmod
+hash -p "$coreutils/bin/cut" cut
+hash -p "$coreutils/bin/echo" echo
+hash -p "$coreutils/bin/head" head
+hash -p "$coreutils/bin/mktemp" mktemp
+hash -p "$coreutils/bin/rm" rm
+hash -p "$coreutils/bin/stat" stat
+hash -p "$coreutils/bin/wc" wc
+hash -p "$file/bin/file" file
+hash -p "$findutils/bin/find" find
+hash -p "$findutils/bin/xargs" xargs
+hash -p "$patchelf/bin/patchelf" patchelf
+hash -p "$ripgrep/bin/rg" rg
+
+# Copy the contents of the original package so we can patch them.
+cp -R "$pkg" "$out"
+
+# Find the new linker that we'll patch into all of the package's executables as
+# the interpreter.
+interp="$(find "$glibc/lib" -type f -maxdepth 1 -executable -name 'ld-linux-*.so*' | head -n1)"
+readonly interp
+
+patch() {
+	declare -r binary="$1" # ELF binary to patch
+	declare -r type="$2"   # "exe" or "lib"
```

Comment:
IIUC we only need this for `exe` files, since the linker-loader will cache the glibc-lib paths even when resolving NEEDED for any `lib` that are encountered later.

Was there a specific reason to include `lib` as well, or more for consistency?

Reply:
Good point. I forgot that we decided to skip the libraries. I'll skip those and re-test.